packages/postgres-database/src/simcore_postgres_database/models/projects_version_control.py

```diff
@@ -0,0 +1,307 @@
+#
+# TODO: create template to produce these tables over another table other than project
+#
+
+import sqlalchemy as sa
+from sqlalchemy.sql import func
+
+from .base import metadata
+from .projects import projects
+from .projects_snapshots import projects_snapshots
+
+# REPOSITORES
+#
+# Projects under version-control are assigned a repository
+#   - keeps information of the current branch to recover HEAD ref
+#   - when repo is deleted, all project_vc_* get deleted
+#
+
+projects_vc_repos = sa.Table(
+    "projects_vc_repos",
+    metadata,
+    sa.Column(
+        "id",
+        sa.BigInteger,
+        nullable=False,
+        primary_key=True,
+        doc="Global vc repo identifier index",
+    ),
+    sa.Column(
+        "project_uuid",
+        sa.String,
+        sa.ForeignKey(
+            projects.c.uuid,
+            name="fk_projects_vc_repos_project_uuid",
+            # ondelete: if project is deleted, this repo is invalidated.
+        ),
+        nullable=False,
+        unique=True,
+        doc="Project under version control"
+        "Used as a working copy (WC) to produce/checkout snapshots.",
+    ),
+    sa.Column(
+        "project_checksum",
+        sa.String,
+        nullable=True,
```

Comment:
the idea is that this column keeps the last checksum of the current working copy or WC  (i.e. the ``project_uuid``).
It is used as a cache mechanism such that we do not need to compute the checksum every time. It is the only field that will be updated by the user so the ``modified`` field might be used to determine whether it is valid or not.
I added some info on the doc

Some scenarios:
- *commit*: compares ``project_vc_repos.project_checksum`` with  HEAD's  ``projects_vc_commit.snapshot_checksum`` to determine if there are changes
- *checkout* : a row of ``project_vc_snapshot`` replaces the ``project_uuid`` WC and the ``projecst_vc_snapshot.checksum`` is copied there.

Reply:
my question here has more to do with: can we add a constraint such that the table does not get null checksum? If I understand correctly, it can only be null when you init the repo. after that there should ALWAYS be a value correct?

Reply:
yes, that is the case.